sgl-pdlb/src/strategy_lb.rs

```diff
@@ -0,0 +1,194 @@
+use rand::Rng;
+use serde_json::json;
+
+#[derive(Debug, Clone)]
+pub enum EngineType {
+    Prefill,
+    Decode,
+}
+
+#[derive(Debug, Clone)]
+pub struct EngineInfo {
+    pub engine_type: EngineType,
+    pub url: String,
+    pub boostrap_port: Option<u16>,
+}
+
+impl EngineInfo {
+    pub fn new_prefill(url: String, boostrap_port: Option<u16>) -> Self {
+        EngineInfo {
+            engine_type: EngineType::Prefill,
+            url,
+            boostrap_port,
+        }
+    }
+
+    pub fn new_decode(url: String) -> Self {
+        EngineInfo {
+            engine_type: EngineType::Decode,
+            url,
+            boostrap_port: None,
+        }
+    }
+
+    pub fn api_path(&self, api_path: &str) -> String {
+        if api_path.starts_with("/") {
+            format!("{}{}", self.url, api_path)
+        } else {
+            format!("{}/{}", self.url, api_path)
+        }
+    }
+
+    pub fn to_string(&self) -> String {
+        format!("({:?}@{})", self.engine_type, self.url)
+    }
+
+    pub fn get_hostname(&self) -> String {
+        let url = self
+            .url
+            .trim_start_matches("http://")
+            .trim_start_matches("https://");
+        url.split(':').next().unwrap().to_string()
+    }
+}
+
+pub struct EngineLoad {
+    pub engine_info: EngineInfo,
+    pub load: isize,
+}
+
+impl EngineLoad {
+    pub fn from_json(engine_info: &EngineInfo, json: &serde_json::Value) -> Self {
+        let load = match json.get("load") {
+            Some(load) => load.as_i64().unwrap_or(-1) as isize,
+            None => -1,
+        };
+        EngineLoad {
+            engine_info: engine_info.clone(),
+            load,
+        }
+    }
+    pub fn to_json(&self) -> serde_json::Value {
+        json!({
+            "engine": self.engine_info.to_string(),
+            "load": self.load,
+        })
+    }
+
+    pub fn to_string(&self) -> String {
+        format!("{}: {}", self.engine_info.to_string(), self.load)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum LBPolicy {
+    Random,
+    PowerOfTwo,
+}
+
+#[derive(Debug, Clone)]
+pub struct StrategyLB {
+    pub policy: LBPolicy,
+    pub prefill_servers: Vec<EngineInfo>,
+    pub decode_servers: Vec<EngineInfo>,
+}
+
+impl StrategyLB {
+    pub fn new(
+        policy: String,
+        prefill_infos: Vec<(String, Option<u16>)>,
```

Comment:
nit: I personally love to use `struct PrefillInfo { url: String, port: uOption<u16>}` but it is just nits